### PR TITLE
Improve linux build docs

### PR DESCRIPTION
**Pull request by nyurik**
* ensure docker image uses host's files -- much better for IDE and other regular work
* ensure docker does not create files as `root` -- harder to cleanup
* improve docs

### CODE REVIEW

File: platform/linux/README.md [hunk 1]
```diff
@@ -2,77 +2,85 @@
...
+
+![Sample image of world from mbgl-render command](/misc/sample-maplibre-style-mbgl-render-out.png)
+
+### Outside of Docker
+
+Render output image using default MapLibre demo tiles server. You can run this binary on your host machine even if you built it inside a Docker container.
 
 ```bash
 ./build/bin/mbgl-render --style https://raw.githubusercontent.com/maplibre/demotiles/gh-pages/style.json --output out.png
+open out.png
```
Comment (louwers) [inline platform/linux/README.md:69]: This is not a standard program as far as I am aware, maybe use `xdg-open`.

**Review by louwers — APPROVED** [review]

File: platform/windows/README.md [hunk 2]
```diff
@@ -83,7 +83,7 @@ If all went well and target `mbgl-render` or `ALL_BUILD` was chosen, there shoul
 
 This should produce an `out.png` map tile image with the default MapLibre styling from [the MapLibre demo](https://maplibre.org/).
 
-![Sample image of world from mbgl-render command](/misc/sample-maplibre-style-mbgl-render-out.png)
+![Sample image of world from mbgl-render command](https://github.com/user-attachments/assets/b1a9355f-731e-4a0f-ab90-68c5ae4beb65)
```
Comment (nyurik) [inline platform/windows/README.md:86]: I don't think this is a good path: repo's content should not depend on external services like this. When all content is together, cloning it is enough -- and we can easily do any kind of markdown book packaging to create a site. When it's external, it can't not be part of the docs.  You could do this as a separate PR and discuss it there of course.

Comment (louwers) [reply platform/windows/README.md]: I reverted the change.
